### DISCUSSION TITLE
Important for those using StageOrientationEvent.

### DISCUSSION THREAD

**Discussion by manishpatel0404** [airsdk/Adobe-Runtime-Support#1892] [General]
I just had a harrowing experience with the `setOrientation` method and `StageOrientationEvent.ORIENTATION_CHANGE` event.

According to [AS3 Reference](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/events/StageOrientationEvent.html)
`Note: If the autoOrients property is false, then the stage orientation does not change when a device is rotated. Thus, StageOrientationEvents are only dispatched for device rotation when autoOrients is true.`

First, an honest admission: I had not read this line, when I implemented the method and event.
I had always set `<autoOrients>false</autoOrients>` and was merrily using the `setOrientation` method and `StageOrientationEvent.ORIENTATION_CHANGE` combination to switch my UI from PORTRAIT to LANDSCAPE and vice versa. Despite what the reference docs above, all was going hunky dory. All the way upto Android 11. And then, Android 12 users started complaining about in their words, "screen display issues".

Upon investigating the matter further in debug mode, I learnt the following:

1. `StageOrientationEvent.ORIENTATION_CHANGE` fires even in Android 12 (regardless of whether `autoOrients` is set to `true` or `false`)
2. Up until Android 11, the `stage` reported the changed dimensions, regardless of how frequently you switched the orientation.
3. In Android 12, the `stage` did not report the correct dimensions after switching orientation. In my case, switching from PORTRAIT (default) to LANDSCAPE (rotatedRight), `stage.stageWidth` and `stage.stageHeight` still reported PORTRAIT dimensions inside the `StageOrientationEvent.ORIENTATION_CHANGE` handler (which proves that at least the event is fired).

As a workaround, I used `Event.RESIZE` to listen to the changed orientation (not very intuitive, but it works). This works on older versions as well as Android 12. I am going to publish the update and rest the case.

NOTE: I did not use `Stage.autoOrients` property. Instead, I set this in the application manifest, thinking it was a good "one place" to set this. I don't know if there is a disconnect between the application manifest and the stage property, or I am missing a point, or it is a bug. Maybe, @ajwfrost can investigate this further and throw some light on the matter.

**Comment by MalacTheLittle** [comment]
I had some issues with `StageOrientationEvent.ORIENTATION_CHANGE` few years ago and since then I'm using `Event.RESIZE` and it works in all cases for me. It's easy to determine portrait or landscape by checking max value of stageWidth and stageHeight:

`var screenOrientation:String = stage.stageHeight > stage.stageWidth ? "portrait" : "landscape";`

**Reply by teotigraphix** [reply]
I do the same thing, wonder if it will blow up one day. :)
